code/controlconfig/controlsconfig.cpp

```diff
@@ -2601,17 +2608,21 @@ int check_control_used(int id, int key)
 
 int check_control(int id, int key) 
 {
-	if (check_control_used(id, key)) {
-		if (Ignored_keys[id]) {
-			if (Ignored_keys[id] > 0) {
-				Ignored_keys[id]--;
-			}
-			return 0;
-		}
-		return 1;
+	const bool is_control_used = check_control_used(id, key) != 0;
+	const bool is_ignored = Ignored_keys[id] != 0;
+
+	//Decrement Ignored_keys if key was pressed and is ignored
+	if (is_control_used && Ignored_keys[id] > 0) {
+		Ignored_keys[id]--;
 	}
 
-	if (Control_config[id].continuous_ongoing) {
+	bool control_triggered = is_control_used && !is_ignored;
+
+	//Only call lua in here when it's a continuous button
+	if (Control_config[id].type == CC_TYPE_CONTINUOUS)
+		control_triggered &= !control_run_lua(static_cast<IoActionId>(id), control_triggered ? 1 : 0);
```

Comment:
I'll be doing single comments because I don't know how far I'll get to review before we trigger 22.2 RC phase.

A bitwise operation on a bool makes me nervous even though it's probably fine, and the syntax is confusing.  I had to do a logic table to figure out what the possibilities are.  It turns out that this is equivalent to 

```if ((Control_config[id].type == CC_TYPE_CONTINUOUS) && !control_run_lua(static_cast<IoActionId>(id)), control_triggered ? 1 : 0)) {
		control_triggered = false; 
}```

Because `control_triggered` can only *remain* true if `control_run_lua(...)` returns false, the compiler respects the order that we write our checks, so, aiui, it will not run `control_run_lua(...)` if `Control_config[id].type == CC_TYPE_CONTINUOUS` is false.

Reply:
@BMagnu Need confirmation on the logic here.

  check_control_used()'s return value is the state of the control, that is, wether it was activated or not since the last frame.

  control_run_lua()'s return value is "whether the control should be overridden or not" according to its docs, but my questions are "overridden by what?" and "How does this play into control_check_used()?"

Reply:
A control is overridden, if the Lua override hook returns true.
General logic on most scripting overrides in FSO imply, that something being successfully overridden is as if it had never happened at all. In the case of controls: If you return true from an override hook, behave as if the button has not been pressed.
Hence, an overridden control is, at least as far as the rest of FSO is concerned, not used.

Reply:
So then, the logic is correct.  It's just convoluted. I believe what I put above is an easier way to express the same logic.